packages/next/package.json

```
@@ -123,7 +123,13 @@
   },
   "peerDependencies": {
     "react": "^16.6.0",
-    "react-dom": "^16.6.0"
+    "react-dom": "^16.6.0",
+    "sharp": "0.26.2"
+  },
+  "peerDependenciesMeta": {
+    "sharp": {
+      "optional": true
+    }
```

```suggestion
    "react-dom": "^16.6.0"
  },
  "optionalDependencies": {
    "sharp": "^0.26.2"
  }
```

Let's use `optionalDependencies`.

Also remove it from `devDependencies` since this will install be default.

---

Exactly, it will always be installed, based on the error defined here https://github.com/vercel/next.js/blob/17cddcb92a329c87d1bb05baa62dd6e8d4c1fc9a/packages/next/next-server/server/image-optimizer.ts#L157-L171 and here https://github.com/vercel/next.js/blob/17cddcb92a329c87d1bb05baa62dd6e8d4c1fc9a/errors/install-sharp.md that is not what you want

https://yarnpkg.com/configuration/manifest#optionalDependencies

---

We can probably remove that message after changing to `optionalDependencies`

---

@styfle This doesn't look great. Next.js is very bloated package.
https://packagephobia.com/result?p=next@10.0.0